### PR TITLE
🧪 Scout: add unit tests for element types and nesting edge cases

### PR DESCRIPTION
**Pull request by cungminh2710**
💡 What: Added unit tests for ICU AST element types and specific parsing edge cases.
🎯 Why: To ensure structural consistency and protect ICU escaping and nested pound sign resolution.
🧩 Scope: internal/i18n/icuparser/ast_test.go, internal/i18n/icuparser/parse_test.go
✅ Verification: make fmt, make lint, and go test ./...
🛡️ Confidence: High, as these tests cover previously untested structural and parsing logic.

---
*PR created automatically by Jules for task [1374574287915054763](https://jules.google.com/task/1374574287915054763) started by @cungminh2710*

### CODE REVIEW

File: internal/i18n/icuparser/parse_test.go [hunk 1]
```diff
@@ -744,3 +744,66 @@ func TestParseASTSelectOrdinal(t *testing.T) {
...
+	// Inside the select branch, # should still refer to the plural argument 'n'.
+	msg := "{n, plural, other {{gender, select, male {# he} other {# they}}}}"
+	elems, err := Parse(msg, nil)
+	if err != nil {
+		t.Fatalf("Parse failed: %v", err)
+	}
+
+	pl := elems[0].(PluralElement)
+	opt := pl.Options[0] // other
+	sel := opt.Value[0].(SelectElement)
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/icuparser/parse_test.go]: @jules fix
